nodes/persistence-node/src/services/UnresponsiveEnsNodeProcessor.ts

```diff
@@ -0,0 +1,80 @@
+import { ethers } from "ethers";
+import { getIpfsHashFromContenthash } from "../getIpfsHashFromContenthash";
+import { sleep } from "../sleep";
+import { toShortString } from "../toShortString";
+import { ProcessEnsIpfsResult } from "../types/ProcessEnsIpfsResult";
+import { Storage } from "../types/Storage";
+import { CacheRunner } from "./CacheRunner";
+import { Logger } from "./Logger";
+
+interface IDependencies {
+  storage: Storage,
+  cacheRunner: CacheRunner,
+  logger: Logger,
+  ensPublicResolver: ethers.Contract,
+}
+
+export class UnresponsiveEnsNodeProcessor {
+  isCanceled = false;
+  deps: IDependencies;
+  
+  constructor(deps: IDependencies) {
+    this.deps = deps;
+  }
+
+  async run() {
+    this.deps.logger.log("Processing unresponsive packages...");
+
+    while (true) {
+      const unresponsiveEnsNodes = Object.keys(this.deps.storage.unresponsiveEnsNodes);
+            
+      if (this.isCanceled && !unresponsiveEnsNodes.length) {
+        this.deps.logger.log("Processing of unresponsive packages cancelled");
+        return;
+      }
+            
+      if (!unresponsiveEnsNodes.length) {
+        await sleep(500);
+        continue;
+      }
+
+      await this.processNodes(unresponsiveEnsNodes);
+    }
+  }
+
+  async processNodes(ensNodes: string[]) {
+    for (const ensNode of ensNodes) {
+      try {
+        // FIX ME: once we introduce exponential backoff logic in retries, 
```

Comment:
In the future, use "TODO: " e.g "TODO: FIX ME: once ...." or "TODO: once ..." both are fine
It's easier to keep this practice and then search at any point the "TODO" comments :)